packages/lib/src/interactions/InteractionsProvider.tsx

```diff
@@ -1,18 +1,15 @@
 import type { ReactNode } from 'react';
 import { createContext, useCallback, useContext, useState } from 'react';
 
-import type { InteractionEntry, ModifierKey, MouseButton } from './models';
+import { Interaction } from './interaction';
+import type { InteractionConfig } from './models';
 
 export interface InteractionsContextValue {
-  registerInteraction: (id: string, value: InteractionEntry) => void;
```

Comment:
The second parameter is no longer stored "as is" in the interactions map below. It is now a config object used to construct an instance of the `Interaction` class.